pkg/cli/admin/nodeimage/create.go

```diff
@@ -116,17 +138,36 @@ type CreateOptions struct {
 	AssetsDir string
 	// OutputName allows the user to specify the name of the generated image.
 	OutputName string
+	// Simpler interface for creating a single node
+	SingleNodeOpts *singleNodeCreateOptions
 
 	nodeJoinerExitCode int
 	rsyncRshCmd        string
 }
 
+type singleNodeCreateOptions struct {
+	MacAddress      string
+	CPUArchitecture string
+	SSHKey          string
+	Hostname        string
+	RootDeviceHints string
+	NetworkConfig   string
+}
+
 // AddFlags defined the required command flags.
 func (o *CreateOptions) AddFlags(cmd *cobra.Command) {
 	flags := o.addBaseFlags(cmd)
 
 	flags.StringVar(&o.AssetsDir, "dir", o.AssetsDir, "The path containing the configuration file, used also to store the generated artifacts.")
 	flags.StringVarP(&o.OutputName, "output-name", "o", "node.iso", "The name of the output image.")
+
+	flags.StringP(snFlagMacAddress, "m", "", "Single node flag. MAC address used to identify the host to apply the configuration. If specified, the nodes-config.yaml config file will not be used.")
+	usageFmt := "Single node flag. %s. Valid only when `mac-address` is defined."
+	flags.StringP(snFlagCpuArch, "c", "", fmt.Sprintf(usageFmt, "The CPU architecture to be use to install the node"))
+	flags.StringP(snFlagSshKey, "k", "", fmt.Sprintf(usageFmt, "The SSH key used to access the node"))
```

Comment:
Does the key have to be encoded or formatted in a certain way to be specified as a single line string?

Reply:
My assumption was this will be the filepath instead of putting the actual key). If its a filepath, please update the message text.

Reply:
It's exactly the same of https://github.com/openshift/installer/blob/a651e1a49ceafcb5cbe6f6244e315a2381013aab/data/data/install.openshift.io_installconfigs.yaml#L4677-L4680 (note: it's not a path).
The node-joiner applies the standard ssh key validation, a format example it's shown in this [test](https://github.com/openshift/installer/blob/a651e1a49ceafcb5cbe6f6244e315a2381013aab/pkg/asset/agent/joiner/addnodesconfig_test.go#L39)

Reply:
Let's read the key from a file path passed as an arg here.
Everybody already has their SSH public key in a file.
Nobody wants to deal with a 400 character-long field in the middle of a CLI command.
Picking a sensible default will require us to read it from the file anyway, so this makes it simpler as well.